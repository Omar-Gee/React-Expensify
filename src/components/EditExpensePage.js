import React from 'react';

const EditExpensePage = () => (
	<div>
		This is from This is from the edit page
	</div>
);

export default EditExpensePage;