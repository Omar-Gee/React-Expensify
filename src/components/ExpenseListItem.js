import React from 'react';
import { removeExpense } from '../actions/expenses';
import { connect } from 'react-redux';

const ExpenseListItem = ({id, description, amount, createdAt, dispatch}) => (
	<div>
		<h3>{description}</h3>
		<p>Amount: {amount}</p>
		<p>CreatedAt: {createdAt}</p>
		<button onClick={ () => {
			dispatch(removeExpense({id}))		
		}}
		>
			Remove
		</button>
	</div>
);

// const mapStateToProps = (state) => {
// 	return {	
// 		expenses: state.expenses
// 	}
// }

export default connect()(ExpenseListItem);