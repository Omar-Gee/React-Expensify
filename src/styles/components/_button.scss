.button {
	margin: 0;
	color: white;
	background-color: $soft-blue;
	display: inline-block;
	padding: $s-size;
	border: none;
	font-weight: 300;
	font-size: $font-size-large;
	text-decoration: none;
	line-height: 1;
}

.button--link{
	background: none;
}
