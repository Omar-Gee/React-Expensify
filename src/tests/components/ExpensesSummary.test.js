import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary, ExpensesSummary } from '../../components/ExpensesSummary';

test('should give ExpensesSummary component with 1 expense', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>)
	expect(wrapper).toMatchSnapshot();
});

test('should give ExpensesSummary component with multiple expenses', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={2353252335566}/>)
	expect(wrapper).toMatchSnapshot();
});