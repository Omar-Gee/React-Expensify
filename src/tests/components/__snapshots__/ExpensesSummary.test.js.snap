// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should give ExpensesSummary component with 1 expense 1`] = `
<div>
  <h3>
    viewing 
    1
     
    expense
     totalling 
    $2.35
  </h3>
</div>
`;

exports[`should give ExpensesSummary component with multiple expenses 1`] = `
<div>
  <h3>
    viewing 
    23
     
    expenses
     totalling 
    $23,532,523,355.66
  </h3>
</div>
`;
